src/hooks/useCharactersApi.ts

```diff
@@ -31,27 +32,28 @@ const useCharactersApi = (searchParams) => {
     [searchParams],
   );
 
-  const { data, error, setSize, isValidating } = useSWRInfinite(
+  const { data, error, setSize, isValidating } = useSWRInfinite<IData, Error>(
     getKey,
     fetcher,
   );
 
   useEffect(() => {
-    if (!data) return;
-
+    if (!data?.[0]) return;
     if (data[0].error) {
       setCharacters([]);
       setHasNextPage(false);
       return;
     }
 
-    const allCharacters = data.flatMap((data) => data.results);
-    const count = data[0]?.info.count;
-    const nextPage = data[data.length - 1]?.info.next;
+    const allCharacters = data
+      .flatMap((data: IData) => data.results)
+      .filter((character): character is ICharacter => character !== undefined); //Filtering ensures that the allCharacters array contains only elements of type Character.
+    const count = data[0].info?.count ?? 0; //We use the "zero coalesce" operator (??) to set the default value to 0 if count is undefined.
+    const nextPage = data[data.length - 1].info?.next;
```

Comment:
[nth] `nextPage` could be checked with `Boolean` in the initiation since there is no other usage
```suggestion
    const nextPage = Boolean(data[data.length - 1].info?.next);

    setCharacters(allCharacters);
    setTotalCount(count);
    setHasNextPage(nextPage);
```